starters/serverless-framework-sqs-dynamodb/src/handlers/technology_update.ts

```diff
@@ -0,0 +1,34 @@
+import type { APIGatewayProxyHandler } from 'aws-lambda';
+import { ZodError } from 'zod';
+import { StatusCodes } from 'http-status-codes';
+import { TechnologyUpdateSchema } from '@/types/technology';
+import { update as updateTechnology } from '@/models/technology/update';
+import { responseHelper } from '@/utils/responseHelper/responseHelper';
+import { getErrorMessage } from '@/utils/error/getErrorMessage';
+
+export const handler: APIGatewayProxyHandler = async (event) => {
+	const id = event.pathParameters?.id;
+	if (!id) {
+		return responseHelper(StatusCodes.BAD_REQUEST, 'Bad Request: no id provided');
+	}
+
+	try {
+		const payload = event.body ? JSON.parse(event.body) : null;
+		TechnologyUpdateSchema.parse(payload);
+		const technology = await updateTechnology(id, payload);
+		if (!technology) {
+			return responseHelper(StatusCodes.NOT_FOUND, null);
+		}
+
+		return responseHelper(StatusCodes.OK, technology);
```

Comment:
Should this method invalidate the cache entry for the item's id? 🤔 

If the GET technology/:id enpoint is cached for 5 minutes, and then an update occurs, the previous value gets returned back in the GET endpoint. I guess same goes for the DELETE and the CREATE methods as well, namely the caching of the all-technologies endpoint should also be invalidated if that is cached. 🤔

Reply:
I handle it a layer deeper in the `models/technology/update`. it is doing the invalidation but the invalidation is handled by the model class that determines whether a value is from cache or database instead of the handler layer. Separation of concerns was my reasoning.